handler.cpp

```diff
@@ -7,67 +7,82 @@
 }
 
 RequestHandler::Status NotFoundHandler::HandleRequest(const Request& request, Response* response){
+    std::string body = "404 NOT FOUND";
+
     response->SetStatus(Response::NOT_FOUND);
-    response->AddHeader("Content-Length", std::to_string(request.raw_request().size() - 4));
+    response->AddHeader("Content-Length", std::to_string(body.size()));
     response->AddHeader("Content-Type", "text/plain");
-    response->SetBody("404 NOT FOUND");
-    return RequestHandler::PASS; //should this be FAIL actually?
+    response->SetBody(body);
+    return RequestHandler::PASS;
 }
 
+//////////////////////////////////////////////////////////////////////////////////////
+/////////////////////////////////// Echo Handler /////////////////////////////////////
+//////////////////////////////////////////////////////////////////////////////////////
+
 RequestHandler::Status EchoHandler::Init(const std::string& uri_prefix, const NginxConfig& config) {
     return RequestHandler::PASS;
 }
 
 RequestHandler::Status EchoHandler::HandleRequest(const Request& request, Response* response) {
-    std::cout << "In EchoHandler" << std::endl;
     response->SetStatus(Response::OK);
     response->AddHeader("Content-Length", std::to_string(request.raw_request().size() - 4));
     response->AddHeader("Content-Type", "text/plain");
     response->SetBody(request.raw_request());
-
     return RequestHandler::PASS;
 }
 
+//////////////////////////////////////////////////////////////////////////////////////
+///////////////////////////////// Static Handler /////////////////////////////////////
+//////////////////////////////////////////////////////////////////////////////////////
+
 // Sets dir_from_config to directory specified after root in config
 RequestHandler::Status StaticHandler::Init(const std::string& uri_prefix, const NginxConfig& config) {
     for (unsigned int i = 0; i < config.statements_.size(); i++){
-      if (config.statements_[i]->tokens_[0] == "root"){
-        dir_from_config = config.statements_[i]->tokens_[1];
-      }
+        // get the root from the child block
+        if (config.statements_[i]->tokens_[0] == "root" && config.statements_[i]->tokens_.size() == 2){
+            serve_path = config.statements_[i]->tokens_[1];
+        }
     }
+
     return RequestHandler::PASS;
 }
 
-
+// Attempts to serve file that was requested by the request object
+// If not possible, we return fail signal for outside class to handle
+// calling not found handler
 RequestHandler::Status StaticHandler::HandleRequest(const Request& request, Response* response) {
-    std::cout << "========= Handling Static =========" << std::endl;
-    url = request.uri();
-    std::string abs_path = get_exec_path() + dir_from_config + "/" + url;
-    std::cout << "Serving file from: " << abs_path << std::endl;
+    std::string abs_path = get_exec_path() + serve_path + "/" + request.uri();
+    std::cout << "Attempting to serve file from: " << abs_path << std::endl;
```

Comment:
what is the point of this stuff? don't we need to just stick the file name to the end of the root name?